src/site/cpp-objects.rst

```diff
@@ -169,6 +169,66 @@ Using the Friend Class
 ContactList
 ===========
 
+* We will create a ``ContactList`` class to keep track of ``Friend`` objects
+* The ``ContactList`` will know it's
+    * Capacity (arrays do not know their capacity)
+    * Size (only one we would want access to)
+    * An array to hold to keep track of the ``Friend`` objects
+
+* We will also want
+    * Constructors
+    * A way to add ``Friend`` objects
+    * A way to remove ``Friend`` objects
+    * Access the size
+    * A way to update a ``Friend`` object's email
+    * A linear search
+    * A way to access a ``Friend`` from a specific intex
+
+
+Header
+------
+
+* Below is a complete ``ContactList.h``
+* Notice that the only private variable with an accessor is ``size_``
+* You will also notice the *deconstructor* ``~ContactList();``
+    * Simply put, deconstructors describe how to deallocate the memory assigned to the instance
+    * This will be discussed in more detail in the implementation section below
+
+* And you will also notice the *copy constructor* ``ContactList(const ContactList &contactList)``
+    * Simply, this describes how to make a copy of an instance of this class
+    * More on this in the implementation section
+
+.. code-block:: cpp
+    :linenos:
+
+    // ContactList.h
+    #include "Friend.h"
+
+    class ContactList {
+        private:
+            const int DEFAULT_CAPACITY = 10;
+            const int NOT_FOUND = -1;
+            int size_;
+            int capacity_;
+            Friend** friends_;
+
+            void expandCapacity();
+
+        public:
+            ContactList();
+            ContactList(int);
+            ContactList(const ContactList& contactList);
+            ~ContactList();
+
+            const int& size() const {return size_;}
+            void add(std::string, std::string, std::string);
+            void remove(std::string, std::string, std::string);
+            void updateEmail(std::string, std::string, std::string);
+            int find(std::string, std::string, std::string);
+            Friend* at(int);
+            const Friend* operator[](std::size_t idx) const { return friends_[idx]; }
```

Comment:
This doesn't really make sense when the class is non-const. You're also saying the thing being pointed to is const, not that the pointer is itself const. The pointer being const is irrelevent, as you return a copy of it anyways. What the caller does with a copy is up to them.

At the end of the day, this signature makes sense for a const class instance, but you need the non-const overload which returns a non-const `Friend`.

Reply:
Following [this](https://en.cppreference.com/w/cpp/language/operators), there is typically 2:

```cpp
          value_t& operator[](std::size_t idx)       { return mVector[idx]; }
    const value_t& operator[](std::size_t idx) const { return mVector[idx]; }
```

Or, more specifically, in this case:

```cpp
              Friend* operator[](std::size_t idx)       { return friends_[idx]; }
        const Friend* operator[](std::size_t idx) const { return friends_[idx]; }
```

All that said, based on the changes to the ``Friend`` and ``ContactList`` example in Java, the cpp versions of the classes will need to change quite a bit. 

This will stay as draft until I get this figured out.

Reply:
That is what I was looking for. From your link and seeing cpp again, it does make sense to return references rather than pointers, unless the aim is to reproduce java first? I forget. At any rate, references are all over the standard library containers, they do make sense for containers like `ContactList` here. `operator[]` gives you the literal element, not a pointer to it.